app/assets/javascripts/components/option-select.js

```diff
@@ -56,21 +60,30 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     }
 
     // Attach listener to update checked count
-    this.$optionSelect.on('change', "input[type='checkbox']", this.updateCheckedCount.bind(this))
+    for (var j = 0; j < this.$options.length; j++) {
+      var anotherThis = this
+      this.$options[j].addEventListener('change', function () {
```

Comment:
Good shout!  Yeah, this does seem to work nicely.  

I did have to make a choice though - because of the test data in `optionSelectWithAttrs()`. 

- I could target everything below `this.$optionsContainer` - because it was the lowest element in the test data, 
- I could add an `if (ulElement) { ... }` guard around the `addEventListener` block in the code, or
- I could add the elements inside `<div class="app-c-option-select__container js-options-container">` down to `<ul class="govuk-checkboxes gem-c-checkboxes__list">` into the `optionSelectWithAttrs()` method in the spec file.

I know these additional elements aren't required in the tests, but I felt targeting everything below `this.$optionsContainer` or adding a guard clause (just so the tests would pass) wasn't the right approach.  And as I wanted to get as close to the `<li>'s` as possible, I chose the second option. 

**BTW**: While I was working on this, I found that I was able to move the `var that = this` line higher in the code, eliminate the duplicate `that = this` lines, and use `that` inside the two event listeners (lines 63 and 75 now).  This appears to work ok - the tests pass and the preview works as expected.  I know `this` is a tricky customer in JS, so please shout if this won't work.

Reply:
Sounds sensible. Generally I try to avoid doing anything to the production code just to make a test pass, but sometimes that's hard.

Reply:
Sorry, my bad - it's the 3rd option - the last one I went with. This does avoid adding anything to the production code.